clang/test/CodeGenSYCL/esimd-private-global.cpp

```diff
@@ -1,14 +1,27 @@
+// TODO: previously code generation and ESIMD lowering was
+// a part of the same %clang_cc1 invocation, but now it is
+// separate. So, we can split this test into 2, where one
+// will be testing code generation and the second ESIMD lowering.
+//
 // RUN: %clang_cc1 -disable-llvm-passes -triple spir64-unknown-unknown-sycldevice \
-// RUN:   -fsycl -fsycl-is-device -fsycl-explicit-simd -emit-llvm %s -o - | \
-// RUN:   FileCheck %s
+// RUN:   -fsycl -fsycl-is-device -fsycl-explicit-simd -emit-llvm %s -o %t
+// RUN: sycl-post-link -split-esimd -lower-esimd -O0 -S %t -o %t.table
+// RUN: FileCheck %s -input-file=%t_esimd_0.ll
 
 // This test checks that FE allows globals with register_num attribute in ESIMD mode.
 
 __attribute__((opencl_private)) __attribute__((register_num(17))) int vc;
 // CHECK: @vc = {{.+}} i32 0, align 4 #0
 
-SYCL_EXTERNAL void init_vc(int x) {
-  vc = x;
-  // CHECK: store i32 %{{[0-9a-zA-Z_]+}}, i32* @vc
```

Comment:
Can you explain the changes in this test?

Reply:
Sure. The `sycl-post-link` tool doesn't work with `SYCL_EXTERNAL` functions yet. There are `TODO` comments in sycl-post-link.cpp about it. This is why I converted `SYCL_EXTERNAL` function to a kernel to overcome this constraint.

In any case, later we need to split this test and use `opt` instead of `sycl-post-link` to test ESIMD lowering.

Reply:
Thanks.

Reply:
@DenisBakhvalov, since the change removes the "guard" for the SYCL_EXTERNAL feature, can you please file an issue for this missing post-link functionality?

NIT: also I'd suggest to still have a test with SYCL_EXTERNAL, marked as XFAIL for now (and referencing the issue).